libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallChartColoring.ts

```diff
@@ -0,0 +1,45 @@
+// (C) 2023 GoodData Corporation
+import {
+    ColorStrategy,
+    IColorMapping,
+    ICreateColorAssignmentReturnValue,
+    getColorFromMapping,
+    isValidMappedColor,
+} from "@gooddata/sdk-ui-vis-commons";
+import { IColorDescriptor, IColorPalette } from "@gooddata/sdk-model";
+import { DataViewFacade, IColorAssignment } from "@gooddata/sdk-ui";
+
+export class WaterfallChartColorStrategy extends ColorStrategy {
+    protected createColorAssignment(
+        colorPalette: IColorPalette,
+        colorMapping: IColorMapping[],
+        _viewByParentAttribute: any,
+        _viewByAttribute: any,
+        dv: DataViewFacade,
+    ): ICreateColorAssignmentReturnValue {
+        const defaultColorPaletteItems: IColorDescriptor[] = [
+            "properties.color.total",
+            "properties.color.positive",
+            "properties.color.negative",
+        ].map((id) => ({
+            colorHeaderItem: { id, name: id },
+        }));
```

Comment:
`defaultColorPaletteItems` could be extracted from this class as it is not called every time.

Reply:
I updated this. Thank you!